### PR TITLE
Env overlay correction

### PR DESCRIPTION
**Pull request by marinagmoreira**
To get the unit tests working I had to switch the order of the catkin_package declaration in the cmake files, turns out that this interferes with the environment overlay for isaac. The problematic tests were converted to rostest (still gtest) to solve the problem and allow correct order in the cmake.

### CODE REVIEW

File: localization/camera/CMakeLists.txt [hunk 1]
```diff
@@ -46,16 +53,26 @@ create_tool_targets(DIR tools
...
-)
+  add_rostest_gtest(test_camera_model
+    test/test_camera_model.test
+    test/test_camera_model.cc
+  )
+  target_link_libraries(test_camera_model
+    camera
+  )
+
+  find_package(rostest REQUIRED)
```
Comment (bcoltin) [inline localization/camera/CMakeLists.txt:67]: Why are there multiple find_package rostests? Our standard has been to import most things in the top level CMakeLists.txt, could we move this there?

File: localization/interest_point/CMakeLists.txt [hunk 2]
```diff
@@ -50,17 +57,19 @@ create_tool_targets(DIR tools
...
-)
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES interest_point
-  CATKIN_DEPENDS
-#  DEPENDS system_lib
-)
+if(CATKIN_ENABLE_TESTING)
+  find_package(rostest REQUIRED)
```
Comment (bcoltin) [inline localization/interest_point/CMakeLists.txt:62]: same

File: localization/sparse_mapping/CMakeLists.txt [hunk 3]
```diff
@@ -60,16 +66,54 @@ if (NOT USE_CTC)
     EXCLUDE ${EXCLUDE})
 endif (NOT USE_CTC)
 
-create_test_targets(DIR test
-  LIBS ${LIBS} sparse_mapping
-  INC ${INCLUDES}
-  DEPS ${DEPS})
+if(CATKIN_ENABLE_TESTING)
+  find_package(rostest REQUIRED)
```
Comment (bcoltin) [inline localization/sparse_mapping/CMakeLists.txt:70]: same

Comment (bcoltin) [reply localization/sparse_mapping/CMakeLists.txt:70]: lots of these in this file?

File: shared/ff_common/CMakeLists.txt [hunk 4]
```diff
@@ -17,20 +17,29 @@
...
 create_library(TARGET ff_common
   INC ${GFLAGS_INCLUDE_DIRS} ${GLOG_INCLUDE_DIRS}
   ${CMAKE_CURRENT_BINARY_DIR}/include
   LIBS ${GFLAGS_LIBRARIES} ${GLOG_LIBRARIES})
 
-create_test_targets(DIR test
-  LIBS ff_common
-  DEPS ff_common)
+if(CATKIN_ENABLE_TESTING)
+  find_package(rostest REQUIRED)
```
Comment (bcoltin) [inline shared/ff_common/CMakeLists.txt:32]: same here

**Review by bcoltin — CHANGES_REQUESTED** [review]
Can we move "find_package(rostest REQUIRED)" into top CMakeLists.txt and only call it once? Then I think it is good.

File: localization/camera/CMakeLists.txt (hunk 1 above)
Comment (marinagmoreira) [reply localization/camera/CMakeLists.txt]: Thanks for noticing the repeated includes! Definitely a copy paste error.
I don't think we have the package rostest when cross-compiling, that's why it is inside the CATKIN_ENABLE_TESTING flag. This is a convention across all unit tests in the repo.

**Review by bcoltin — APPROVED** [review]

**Pull request merged**